Core/Core/Models/Utilities.cs

```diff
@@ -218,8 +218,9 @@ public static IEnumerable<List<T>> SplitList<T>(List<T> list, int chunkSize = 50
   /// This happens, for example, in the case of multiple display value fallbacks for a given object.
   /// </summary>
   /// <param name="item"> Object to flatten</param>
+  /// <param name="nativeType"> Type to not flat if native type inherited from IEnumerable.</param>
   /// <returns> Flattened objects after to native.</returns>
-  public static List<object> FlattenToNativeConversionResult(object item)
+  public static List<object> FlattenToNativeConversionResult(object item, Type nativeType)
```

Comment:
I Have some comments on this `nativeType` prop.
Why is it needed?

Reply:
I see we're passing `typeof(Entity)`... but Entity doesn't appear to implement `IList`
(assuming this is `Autodesk.AutoCAD.DatabaseServices.Entity`)

Reply:
But on Line 233,
```(current.GetType() != nativeType && current is IList list)``` 

you're checking if the `current.GetType() != nativeType` (which does an exact equality, no subtypes).

Surely changing `IEnumerable` check to `IList` is all that's needed. I don't think this `nativeType` is actually doing anything

Reply:
You are right. Good catch! While testing it I converted it from `IEnumerable` back to `IList` and this change mislead me I guess while having the expected behavior. Fixing it and will reconsider weather we need on `nativeType` or not.

Reply:
changing it to `IEnumerable` was to cover all enumerables. Now keeping it with `IEnumerable` and checking `current` type and `nativeType` with `IsAssignableFrom()` method to get potential parent child relationship to prevent flattening.

Reply:
If we kept to using `IList` or maybe a more general `IReadonlyCollection` interface.

Would that be preferable? Do we need all accept all `IEnumerable` types.
Our conversions should be returning fully enumerated collections (e.g. arrays, Lists, or HashSets)

Reply:
I'm just thinking...

If we were to keep `nativeType` around.
1. I think we could come up with a better name, `IgnoredTypes` for e.g.
2. We should probably make it optional (nullable)
3. Or, make it possible to pass in multiple types (ISet<Type>)

Reply:
I think we do not need to have it `IList`. `IEnumerable` covers all?

1. Agree on better name needed!
2. Why not to have nullable, agree.
3. Might be, do we have any scenario that we might need to pass multiple base types in some connectors? With Rhino, `RhinoObject` should be sufficient but not sure about Revit or others :) - regardless to this question, I'm saying why not to accept multiple types?

Reply:
@JR-Morgan okay, updated it with your suggestion and extended unit tests a bit. Could you double check to make sure all good?